pkg/keycloak/upgrade.go

```diff
@@ -0,0 +1,117 @@
+package keycloak
+
+import (
+	v1 "github.com/openshift/api/apps/v1"
+	cv1 "k8s.io/api/core/v1"
+	"regexp"
+)
+
+func CanUpgrade(version string) bool {
+	// we will handle upgrade for any 7.2.x version
+	r := regexp.MustCompile("^v7.2.*.GA$")
+	return r.MatchString(version)
+}
+
+func DeploymentUpgraded(dc *v1.DeploymentConfig) bool {
+	upgraded := false
+	for _, t := range dc.Spec.Triggers {
+		if t.Type == v1.DeploymentTriggerOnImageChange {
+			if t.ImageChangeParams.From.Name == SSO_IMAGE_STREAM {
+				upgraded = true
+			}
+		}
+	}
+	if !upgraded {
+		return upgraded
+	}
+	upgraded = len(dc.Spec.Template.Spec.Volumes) == 2
+	if !upgraded {
```

Comment:
instead of all of these:
```
if !upgraded {
   return upgraded
}
```
Why not just negate the logic in your check and return false, e.g.
```
if len(dc.Spec.Template.Spec.Volumes) != 2 {
    return false
}
```